src/Browscap/Formatter/JsonFormatter.php

```diff
@@ -128,14 +128,21 @@ private function jsonEncode($val)
         if ($val === null) {
             return '"null"';
         }
+
         if ($val === true) {
             return '"true"';
         }
+
         if ($val === false) {
             return '"false"';
         }
-        if (is_string($val) || is_numeric($val)) {
-            return '"' . addslashes($val) . '"';
+
+        if (is_string($val)) {
+            return json_encode($val);
+        }
+
+        if (is_numeric($val)) {
+            return '"' . json_encode($val) . '"';
```

Comment:
If it's numeric, why do we need the quotes? Same with `true` and `false`...

Why do we need to granularly write the json vs building an array and just doing `json_encode` on the lot?